### PR TITLE
[CALCITE-6725] The caching mechanism key in ElasticsearchSchemaFactor…

### PR DESCRIPTION
**Pull request by AstraHoo**
…y is affected by the order of hosts

### CODE REVIEW

**Comment by AstraHoo** [comment]
@NobiGo Hi,I've squashed the commits. Please review them again.

**Comment by AstraHoo** [comment]
@NobiGo Hi,I've squashed the commits. Please review them again.

**Comment by NobiGo** [comment]
> @NobiGo Hi,I've squashed the commits. Please review them again.

After the CI passes, I will merge it.

File: elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchemaFactory.java [hunk 1]
```diff
@@ -183,6 +186,15 @@ public ElasticsearchSchemaFactory() {
     }
   }
 
+  protected static List<HttpHost> getSortedHost(List<HttpHost> hosts) {
```
Comment (ILuffZhe) [inline elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchemaFactory.java:189]: LGTM!
Nitpick: The results and the following test are described as sortedHosts, I prefer getSortedHosts for this method.I'm still OK if it remains still.

**Review by ILuffZhe — APPROVED** [review]

File: elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchemaFactory.java (hunk 1 above)
Comment (AstraHoo) [reply elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchemaFactory.java:189]: Thanks for your suggestion

**Review by NobiGo — APPROVED** [review]

**Pull request merged**